java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSinkFactory.java

```diff
@@ -85,33 +82,31 @@ public void validate(
             }
         }
 
-        RestHighLevelClientAdapter client;
         // 2. check connection
         try {
-            boolean isConnected;
             if (config.getConnector().equals("elasticsearch")) {
-                client = new ElasticRestHighLevelClientAdapter(host, config);
-                isConnected = client.ping(RequestOptions.DEFAULT);
+                ElasticRestHighLevelClientAdapter esClient =
```

Comment:
Should we wrap the `esClient` with try-with-resource so that when any exception is thrown, we can always remember to close the client. Otherwise, we may forget to close the client and cause resource leak.